tests/unit/iteratorFilter.cpp

```diff
@@ -0,0 +1,385 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <vector>
+#include <numeric>
+#include <iostream>
+
+#include <graphblas.hpp>
+#include <graphblas/utils/iterators/filter.hpp>
+
+
+template< typename FwdIt >
+void testOneOut(
+	const size_t &n, const size_t filtered, const size_t expected,
+	FwdIt &it, const FwdIt &end,
+	grb::RC &rc
+) {
+	rc = grb::SUCCESS;
+	size_t count = 0;
+	for( ; it != end; ++it, ++count ) {
+		if( *it != *(it.operator->()) ) {
+			std::cerr << "Error: dereference operator returns " << *it << ", "
+				<< "but dereferencing the pointer operator returns " << (*(it.operator->()))
+				<< "\n";
+			rc = grb::FAILED;
+			return;
+		}
+		if( *it == filtered ) {
+			std::cerr << "Error: found " << filtered << ", "
+				<< "which should have been filtered out\n";
+			rc = grb::FAILED;
+			return;
+		}
+	}
+	if( count != expected ) {
+		std::cerr << "Error: count should be " << n << ", "
+			<< "but the filtered iterator returned " << count << " elements instead\n";
+		rc = grb::FAILED;
+	}
+}
+
+template< typename FwdIt >
+void testOnlyOne(
+	const size_t &n, const size_t retained, const size_t expected,
+	FwdIt &it, const FwdIt &end,
+	grb::RC &rc
+) {
+	rc = grb::SUCCESS;
+	size_t count = 0;
+	for( ; it != end; ++it, ++count ) {
+		if( *it != *(it.operator->()) ) {
+			std::cerr << "Error: dereference operator returns " << *it << ", "
+				<< "but dereferencing the pointer operator returns " << (*(it.operator->()))
+				<< "\n";
+			rc = grb::FAILED;
+			return;
+		}
+		if( *it != retained ) {
+			std::cerr << "Error: found " << *it << ", "
+				<< "but only value(s) " << retained << " were expected\n";
+			rc = grb::FAILED;
+			return;
+		}
+	}
+	if( count != expected ) {
+		std::cerr << "Error: count should be " << n << ", "
+			<< "but the filtered iterator returned " << count << " elements instead\n";
+		rc = grb::FAILED;
+	}
+}
+
+
+void grb_program( const size_t &n, grb::RC &rc ) {
+	size_t expected;
+	grb::RC local_rc;
+	rc = grb::SUCCESS;
+
+	// first fill some vector v with a range of numbers from 0 to n-1 (inclusive)
+	std::vector< size_t > v( n );
+	std::iota( v.begin(), v.end(), 0 );
+
+	// test for filtering out the number 7
+	{
+		auto itPair = grb::utils::iterators::createIteratorFilter(
+			v.begin(), v.end(),
+			[] (const size_t val) { return val == 7; }
+		);
+		if( n < 8 ) {
+			expected = n;
+		} else {
+			expected = n - 1;
+		}
```

Comment:
```suggestion
		expected = ( n < 8 ) ? n : n-1;
```

Ternary usage in the whole file ?

Reply:
I see it hadn't been committed yet, and I'm too lazy to do this one (the difference in readability is minimal). Hope ok to skip?

Reply:
Absolutely!